cql3/restrictions/restriction.hh

```diff
@@ -74,6 +124,7 @@ protected:
     enum_set<op_enum> _ops;
     target _target = target::SINGLE_COLUMN;
 public:
+    wip::expression expression = false; ///< wip equivalent of *this.
```

Comment:
This is never used, yes? because every subclass writes to this member without reading it (it will be read in merge_with, but only after the subclass wrote it).

The best way to express this is to initialize expression via the constructor. This ensures that you don't ever forget to initialize it.

Reply:
Yes, I'll add it to the constructor signature.  This was useful while I was adding code incrementally.

Reply:
Adding this to the constructor is a lot of work because of all the subclasses.  It is ultimately useless, because the whole class is going away soon.  I'm keeping it as is for now.